include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -387,8 +389,9 @@ inline void multiplexer::do_start_receive()
 inline void multiplexer::discard_message() {
     boost::system::error_code error;
     endpoint_type remote_endpoint;
+    static char empty_buffer_storage;
     next_layer().receive_from
-        ( boost::asio::buffer(static_cast<char*>(nullptr), 0)
+        ( boost::asio::buffer(&empty_buffer_storage, 1)
```

Comment:
@inetic I'll tell you what, I really ought to go do my part by testing the Boost 1.58 beta release anyway, so I'll go add async_wait support and fix up the code to work on Boost < 1.58. Hopefully should have something for tomorrow.